spec/models/app_remote_kill_migration_spec.rb

```diff
@@ -0,0 +1,185 @@
+    FactoryBot.create(
+      :app_remote_kill,
+      app: app,
+      split: feature_gate,
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: "0.9",
+      fixed_version: nil
+    )
+    subject = described_class.new(
+      app: app,
+      split: feature_gate.name,
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: nil,
+      fixed_version: nil
+    )
+
+    expect(subject.save).to eq true
+
+    expect(app.remote_kills).to be_empty
+  end
+
+  it "destroys remote kills with empty string first_bad_version" do
+    FactoryBot.create(
+      :app_remote_kill,
+      app: app,
+      split: feature_gate,
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: "0.9",
+      fixed_version: nil
+    )
+    subject = described_class.new(
+      app: app,
+      split: feature_gate.name,
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: "",
+      fixed_version: nil
+    )
+
+    expect(subject.save).to eq true
+
+    expect(app.remote_kills).to be_empty
+  end
+
+  it "nullifies fixed_version with an empty string for URLEncoded compatibility" do
+    subject = described_class.new(
+      app: app,
+      split: feature_gate.name,
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: "0.9",
+      fixed_version: ""
+    )
+
+    expect(subject.save).to eq true
+
+    result = app.remote_kills.first
+    expect(result.fixed_version).to eq(nil)
+  end
+
+  it "is invalid with no split" do
+    subject = described_class.new(
+      app: app,
+      split: "nonexistant_split",
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: "0.9",
+      fixed_version: ""
+    )
+
+    expect(subject).to have(1).error_on(:split)
+    expect(subject.save).to eq false
+  end
+
+  it "is invalid with no split on delete" do
+    subject = described_class.new(
+      app: app,
+      split: "nonexistant_split",
+      reason: "my_giant_bug_2019",
+      override_to: "false",
+      first_bad_version: nil,
+      fixed_version: nil
+    )
+
+    expect(subject).to have(1).error_on(:split)
+    expect(subject.save).to eq false
+  end
+
+  it "is invalid with no reason" do
+    subject = described_class.new(
+      app: app,
+      split: feature_gate.name,
+      reason: "",
+      override_to: "false",
+      first_bad_version: "1.0",
+      fixed_version: nil
+    )
+
+    expect(subject).to have(1).error_on(:reason)
+    expect(subject.save).to eq false
+  end
+
+  it "is invalid with no reason on delete" do
+    subject = described_class.new(
+      app: app,
+      split: feature_gate.name,
+      reason: "",
+      override_to: "false",
+      first_bad_version: nil,
+      fixed_version: nil
+    )
+
+    expect(subject).to have(1).error_on(:reason)
+    expect(subject.save).to eq false
```

Comment:
Worth checking that this doesn't change the row count?

Reply:
will do!